docs/source/glossary.md

```diff
@@ -0,0 +1,19 @@
+# Glossary
+
+A glossary of common terms used throughout the PyMC3 documentation and examples.
+
+:::::{glossary}
+[Term with external reference](https://www.youtube.com/watch?v=6dc7JgR8eI0)
+  Terms are defined within this glossary directive. The term id is defined as the non
+  indented line, and can be text alone (like {term}`second term`) or also include a link
+  to an external reference.
+
+Second term
+  Definitions can have as many lines as desired, and should be written in markdown. Definitions
+  can contain any markdown formatting for MyST to parse, this includes basic markdown like **bold**
+  as well as MyST directives and roles like {fa}`fort awesome,style=fab`
+Functional Programming
+  Functional programming is writing pure functions
+Dispatching
+  Choosing which implementation of an operation is used
```

Comment:
@ricardoV94 or anyone else who feels like it, please add the right definitions, relevant links... I have left the two example terms and defintions I added originally to show a bit what is possible.